### PR TITLE
feat(v2): Validate basic on the VEs

### PR DESCRIPTION
**Pull request by davidterpay**
### Overview
Adding `validateVoteExtensionsFn` to the V2 implementation of POB. We add this to resolve whether the vote extensions included in a block proposal compose of a supermajority of validators in the network (or other custom fn). 

@nivasan1 Do we need to add one with `context` similar to what we did with Slinky?

### CODE REVIEW

File: abci/proposals.go [hunk 1]
```diff
@@ -165,10 +176,3 @@ func (h *ProposalHandler) ProcessProposalHandler() sdk.ProcessProposalHandler {
 		return cometabci.ResponseProcessProposal{Status: cometabci.ResponseProcessProposal_ACCEPT}
 	}
 }
-
-// RemoveTx removes a transaction from the application-side mempool.
-func (h *ProposalHandler) RemoveTx(tx sdk.Tx) {
```
Comment (nivasan1) [inline abci/proposals.go:170]: Why is this removed?

File: abci/proposals.go [hunk 2]
```diff
@@ -57,12 +54,24 @@ type (
...
+		// according to each lane's verification logic.
 		processLanesHandler blockbuster.ProcessLanesHandler
-		tobLane             TOBLaneProposal
-		logger              log.Logger
-		txEncoder           sdk.TxEncoder
-		txDecoder           sdk.TxDecoder
+
+		// tobLane is the top of block lane which is utilized to verify transactions that
+		// should be included in the top of block.
+		tobLane TOBLaneProposal
```
Comment (nivasan1) [inline abci/proposals.go:71]: discussed offline, was wondering why the tobLane exists independently of the other lanes

File: abci/vote_extensions.go [hunk 3]
```diff
@@ -102,27 +127,59 @@ func (h *VoteExtensionHandler) VerifyVoteExtensionHandler() VerifyVoteExtensionH
...
+				"err", err,
+			)
+
 			h.cache[hash] = err
 			return &ResponseVerifyVoteExtension{Status: ResponseVerifyVoteExtension_REJECT}, err
 		}
 
 		h.cache[hash] = nil
 
+		h.logger.Info(
```
Comment (nivasan1) [inline abci/vote_extensions.go:177]: logging == gud

File: abci/proposal_auction.go [hunk 4]
```diff
@@ -86,6 +86,12 @@ func (h *ProposalHandler) VerifyTOB(ctx sdk.Context, proposalTxs [][]byte) (*Auc
 		return nil, fmt.Errorf("failed to unmarshal last commit info from auction info: %w", err)
 	}
 
+	// verify that the included vote extensions are valid in accordance with the
+	// the preferences of the application
+	if err := h.validateVoteExtensionsFn(ctx, ctx.BlockHeight(), lastCommitInfo); err != nil {
```
Comment (nivasan1) [inline abci/proposal_auction.go:91]: We shld think abt noting down our experiences w/ VEs at present, and thinking abt how the API can be changed to make the dev UX better

**Review by nivasan1 — APPROVED** [review]
comments + questions

File: abci/proposals.go (hunk 1 above)
Comment (davidterpay) [reply abci/proposals.go:170]: removed because its unused wit bb
